src/main/kotlin/org/rucca/cheese/team/TeamMembershipService.kt

```
@@ -0,0 +1,630 @@
+            applicationRepository
+                .findByIdAndTeamIdAndTypeAndStatus(
+                    id = requestId,
+                    teamId = teamId,
+                    type = ApplicationType.REQUEST,
+                    status = ApplicationStatus.PENDING,
+                )
+                .orElseThrow { NotFoundError("Pending request for team", requestId) }
+
+        try {
+            teamService.checkTeamLockingStatus(teamId)
+        } catch (e: TeamLockedError) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)